File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "group_#{params[:group_id]}_chat" # グループごとに別のチャンネルを使う
  end

  def speak(data)
    Chat.create!(message: deta['message'], user: current_user, group_id: params[:group_id])
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from './consumer';

consumer.subscriptions.create({ channel: "ChatChannel", group_id: groupId }, {
  connected() {
    
  },
  
  disconnected() {
    
  },
  
  received(data) {
    $('#messages').append(data['message']);
  },
  
  speak: function(message) {
    return this.perform('speak', { message: message });
  }
});

$(document).on('submit', 'form', function(e) {
  e.preventDefault();
  var message = $('#message').val();
  consumer.subscriptions.chatChannel.speak(message);
  $('#message').val('');
});
 